Melhorias na jornada de LOGIN do app JaChegou.

FabioCristianoLopes19 opened the pull request:
Prezados (as), boa noite.

Melhorias realizada na jornada de LOGIN do app JaChegou.

**Detalhes das melhorias:**

LoginModel: 
- Remoção de comentários.

LoginViewController:
- Melhorias no código;
- Criação de guard let para remoção de opcionais;
- Request Login com o Firebase na LoginViewModel;
- Remoção dos comentários.

LoginViewModel:
- Melhorias no código;
- Ajuste Request Login com o Firebase na LoginViewModel;
- Remoção dos comentários.

RecoverPasswordScreen:
- Melhorias no código;
- Remoção if let viewController.

RecoverPasswordViewModel:
- Melhorias no código;
- Criação guard let para remover opcionais.

RecoverPasswordViewController:
- Melhorias no código;
- Removido do método showAlert da RecoverPasswordScreen (repetição do código);
- Remoção dos comentários.

CreateAccountViewController:
- Melhorias no código;
- Melhorias nos métodos tappedEyePasswordButton e tappedEyeConfirmPasswordButton;
- Criado a classe ValidationData na pasta Utils.
- Remoção dos comentários.

CreateAccountScreen:
- Melhorias no código;
- Remoção dos comentários.

CreateAccountViewModel:
- Melhorias no código;
- Remoção “var user”;
- Remoção dos “prints”e comentários.

Pasta Utils:
- Criado a classe ValidationData.

LoginViewController:
- Ajuste no código (alert).

FirebaseFirestoreManager:
- Melhorias no código;
- Ajuste no código (remoção let newUniqueEvents);
- Remoção dos “prints”e comentários.

* Ajuste realizado no dia 06/nov.
LoginViewModel:
- Melhorias no código, substituindo Closure para Protocolo para manter as boas práticas.

LoginViewController:
- Melhorias no código, substituindo Closure para Protocolo para manter as boas práticas na LoginViewModel.  

Dependências externas:
- Foi realizado o comando pod install para instalar e gerenciar dependências externas no projeto usando o CocoaPods.

Atenciosamente,

FabioCristianoLopes19 committed:
- LoginModel: remoção de comentários.

FabioCristianoLopes19 committed:
LoginViewController:
- Melhorias no código;
- Criação de guard let para remoção de opcionais;
- Request Login com o Firebase na LoginViewModel;
- Remoção dos comentários.

LoginViewModel:
- Melhorias no código;
- Ajuste Request Login com o Firebase na LoginViewModel;
- Remoção dos comentários.

FabioCristianoLopes19 committed:
RecoverPasswordScreen:
- Melhorias no código;
- Remoção if let viewController.

RecoverPasswordViewModel:
- Melhorias no código;
- Criação guard let para remover opcionais.

RecoverPasswordViewController:
- Melhorias no código;
- Removido do método showAlert da RecoverPasswordScreen (repetição do código);
- Remoção dos comentários.

FabioCristianoLopes19 committed:
CreateAccountViewController:
- Melhorias no código;
- Melhorias nos métodos tappedEyePasswordButton e tappedEyeConfirmPasswordButton;
- Criado a classe ValidationData na pasta Utils.
- Remoção dos comentários.

CreateAccountScreen:
- Melhorias no código;
- Remoção dos comentários.

CreateAccountViewModel:
- Melhorias no código;
- Remoção “var user”;
- Remoção dos “prints”e comentários.

Pasta Utils:
- Criado a classe ValidationData.

FabioCristianoLopes19 committed:
LoginViewController:
- Ajuste no código (alert).

GuilhermeBraganca commented on JaChegou/JaChegou/Login/LoginController/ViewModel/LoginViewModel.swift:
```diff
@@ -5,44 +5,42 @@
+import FirebaseAuth
 
+class LoginViewModel {
+    var isLoginButtonEnabled: ((Bool) -> Void)?
```
essa implementação o caio comentou a respeito, vamos deixar assim msm ?

GuilhermeBraganca commented on JaChegou/JaChegou/Login/ResetPasswordController/Screen/RecoverPasswordScreen.swift:
```diff
@@ -48,26 +47,9 @@ class RecoverPasswordScreen: UIView {
         return button
     }()
     
-    @objc func tappedSendEmailButton() {
```
Removeu toda a parte da função do BotãoSendEmailButton. OK. Porém o código foi implementado em outro lugar?

FabioCristianoLopes19 committed:
LoginViewModel:
- Melhorias no código, substituindo Closure para Protocolo para manter as boas práticas.

LoginViewController:
- Melhorias no código, substituindo Closure para Protocolo para manter as boas práticas na LoginViewModel.  Dependências externas:
- Foi realizado o comando pod install para instalar e gerenciar dependências externas no projeto usando o CocoaPods.

FabioCristianoLopes19 committed:
- Melhorias na Jornada de Login.
